File: 프로그래머스/unrated/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.java
import java.util.*;

class Solution {
    public int[] solution(String s) {
        int[] answer = new int[s.length()];
        int[] check = new int[26];
        Arrays.fill(check, -1);
        for(int i = 0; i < s.length(); i++) {
            int tmp = s.charAt(i) - 'a';
            if(check[tmp] == -1) {
                answer[i] = -1;
            } else {
                answer[i] = i - check[tmp];
            }
            check[tmp] = i;
        }
        
        return answer;
    }
}

File: 프로그래머스/1/142086. 가장 가까운 같은 글자/가장 가까운 같은 글자.js
function solution(s) {
    var len = s.length;
    var answer = new Array(len);
    var check = new Array(26).fill(-1);
    for(let i = 0; i < len; i++) {
        let num = s[i].charCodeAt(0) - 'a'.charCodeAt(0);
        if(check[num] === -1) {
            answer[i] = -1;
        } else {
            answer[i] = i - check[num];
        }
        check[num] = i;
    }
    return answer;
}